### PR TITLE
Add Support for Multiplicity Contraints when Creating Policies

### LINKED ISSUE #968: Add Support for Multiplicity Contraints when Creating Policies
# Feature Request

## Description
Mulitiplicity Constraints `AND`, `OR` and `XOR` can not be created using the API-Wrapper yet. There should be an endpoint that supports also creating policies with those constraints.

### PR DESCRIPTION
**Pull request by efiege**
Closes #968 

```[tasklist]
### Checklist
- [x] The PR title is short and expressive.
- [x] I have updated the CHANGELOG.md. See [changelog_update.md](https://github.com/sovity/authority-portal/tree/main/docs/dev/changelog_updates.md) for more information.
- [x] I have updated the Deployment Migration Notes Section in the CHANGELOG.md for any configuration / external API changes.
- [ ] I have updated the Community Edition [Postman-Collection](https://github.com/sovity/edc-extensions/blob/main/docs/api/postman_collection.json) if I changed existing APIs or added new APIs (e.g. for Management-API or API-Wrapper)
- [x] I have performed a **self-review**
```

### CODE REVIEW

File: extensions/wrapper/wrapper-api/src/main/java/de/sovity/edc/ext/wrapper/api/usecase/UseCaseResource.java [hunk 1]
```diff
@@ -59,4 +61,11 @@ List<UiDataOffer> queryCatalog(
         @Valid @NotNull
         CatalogQuery catalogQuery
     );
+
+    @POST
+    @Path("policy-definition")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(description = "Create a new Policy Definition")
+    IdResponseDto createPolicyDefinitionUseCase(PolicyCreateRequest policyCreateRequest);
```
Comment (richardtreier) [inline extensions/wrapper/wrapper-api/src/main/java/de/sovity/edc/ext/wrapper/api/usecase/UseCaseResource.java:70]: The UseCaseApi needs to be accessed separately. The method name will be passed into the API clients as method name, thus the usage would be

`edcClient.useCaseApi().createPolicyDefinitionUseCase(...)`, I believe we can drop the "UseCase" from the method name.

File: extensions/wrapper/wrapper-common-mappers/src/main/java/de/sovity/edc/ext/wrapper/api/common/mappers/utils/AtomicConstraintMapper.java [hunk 2]
```diff
@@ -103,4 +105,16 @@ private AtomicConstraint buildAtomicConstraint(UiPolicyConstraint constraint) {
...
                 .build();
     }
+
+    public AtomicConstraint buildAtomicConstraint(AtomicConstraintDto atomicConstraint) {
+        var left = atomicConstraint.getLeftExpression();
+        var operator = operatorMapper.getOperator(atomicConstraint.getOperator());
+        var right = atomicConstraint.getRightExpression();
+
+        return AtomicConstraint.Builder.newInstance()
+                .leftExpression(new LiteralExpression(left))
```
Comment (richardtreier) [inline extensions/wrapper/wrapper-common-mappers/src/main/java/de/sovity/edc/ext/wrapper/api/common/mappers/utils/AtomicConstraintMapper.java:115]: Comment / Question: Left expression is an `Object`, right? This is a prime example of something that will have to be tested in E2E Tests, as we will only know very far "down the line" X whether the "policies" actually work.

**Review by richardtreier — CHANGES_REQUESTED** [review]

File: extensions/wrapper/wrapper-api/src/main/java/de/sovity/edc/ext/wrapper/api/usecase/UseCaseResource.java (hunk 1 above)
Comment (efiege) [reply extensions/wrapper/wrapper-api/src/main/java/de/sovity/edc/ext/wrapper/api/usecase/UseCaseResource.java:70]: I implemented it like this in the first place, but it seems if there are the same names (also located in different APIs) the `operationId` in the OpenApi spec will be generated to createPolicyDefinition**1**.

`createPolicyDefinition` is also used in the UiApi.

File: extensions/wrapper/wrapper-common-mappers/src/main/java/de/sovity/edc/ext/wrapper/api/common/mappers/utils/AtomicConstraintMapper.java (hunk 2 above)
Comment (efiege) [reply extensions/wrapper/wrapper-common-mappers/src/main/java/de/sovity/edc/ext/wrapper/api/common/mappers/utils/AtomicConstraintMapper.java:115]: Both left and right expressions are of type `String` in the `AtomicConstraintDto` for now. I have not seen more complex `Expressions` to this point.

**Review by richardtreier — APPROVED** [review]
LGTM

**Pull request merged**